### ISSUE TITLE
Unit/army stays invisible even if in vision of stealth units

### ISSUE THREAD

**Issue by LynxAbraxas** [civctp2/civctp2#260]
I once observed that a unit/army stayed invisible even when in vision of a stealth unit and even with owning global sat.
Here the mouse points the tile where the units are not shown (only mentioned in the status-bar):
![ss_2019-11-12_21:25:17](https://user-images.githubusercontent.com/21012234/73598308-e625aa00-4536-11ea-9b43-be109bcaae40.png)
and a Lawyer does not reveal these as well:
![ss_2019-11-12_21:27:25](https://user-images.githubusercontent.com/21012234/73598311-ee7de500-4536-11ea-81b3-cfc275e59495.png)

**Comment by LynxAbraxas** [comment]
Here the corresponding save-game where an Aborigine army (2 fields) east of Chinandega is not visible: 
[dbgVisibility.zip](https://github.com/civctp2/civctp2/files/4143189/dbgVisibility.zip)

**Comment by LynxAbraxas** [comment]
Oddly, this issue is not yet resolved with #314. @williamd67 do you have any ideas why? Also I wonder why with #298 the mouse pointer is not visible in screen shots any more. Is that because it is now drawn by SDL2? Do you know if there would be a switch to have it included in screen shots even if draw with SDL2?

**Comment by williamd67** [comment]
> Oddly, this issue is not yet resolved with #314.

Do you have a new case or reloaded a saved game?
- I do not expect a new case to happen as stacks should move together. In case you have a new case I would appreciate if you can give me a reproduction scenario.
- I do expect it still to happen when the saved game does have incorrect positions for the actor. I can fix that be aligning the actor's position in the serialize method. I will do that in a seperate PR.

**Comment by williamd67** [comment]
> Also I wonder why with #298 the mouse pointer is not visible in screen shots any more.

We discussed it briefly before. It depends what tool you use to take screenshots in case you use `PrtScr`-key the cursor is part of the screenshot. I do notice that in `kmag` the cursor is shown but not as the picture that is shown in CTP2 but as a normal arrow-pointer which does wonder me: what is done by the graphics/windows-manager in this case. Having seen this specific situation I do not expect that a lot can be done. I will do a quick investigation if  there is something mentioned on the internet.

**Comment by LynxAbraxas** [comment]
> Do you have a new case or reloaded a saved game?

Yes, I reloaded the save game. With your comments I figured I should load the save game one turn before and let that run for one turn. It then turns out just as expected:
![ss_2020-06-24_22:52:13](https://user-images.githubusercontent.com/21012234/85626711-be61ea80-b66d-11ea-8fc5-57f39cef7415.png)
Sorry for the fuzz and many thanks for fixing this with #314!
